scripts/ci/style/pyright_check.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+name=$(basename "$0")
+
+source ./scripts/ci/helpers.sh
+
+set_on_failed_callback "FAIL pyright errors"
+
+if [[ -n $GITHUB_BASE_REF ]]; then
+  echo "Running pyright on changed files..."
+  git fetch origin "$GITHUB_BASE_REF"
+  if ! (git diff --name-only --diff-filter=d "origin/$GITHUB_BASE_REF" -z -- '*.py' | xargs -0 --no-run-if-empty pyright); then
+    FAIL "pyright failed on changed files."
+    exit 1
+  fi
+else
+  echo "Running pyright..."
+  pyright . | tee /tmp/"$name"_full
+  if [ -s /tmp/"$name"_full ]; then
+    FAIL "pyright failed."
+    cat /tmp/"$name"_full
+    exit 1
+fi
```

Comment:
pyright seems to output a bunch of stuff even when the lint passes, so this doesn't seem quite right to me; I'd think using exit code in general is cleaner than parsing output, too.

Also, since you're `tee`ing and not redirecting above isn't the `cat` unnecessary?

Reply:
ye true